### PR TITLE
Switching to use new TOC style and organization from SCDF docs

### LINKED ISSUE #208: Adapt to new reference guide UX
As a developer, I'd like to adapt to the new reference guide UX. 

**Acceptance:**
Review "common" doc includes to make sure unrelated platform specific topics aren't included.

### LINKED ISSUE #209: Update docs to include production deployment instructions
As a developer, I'd like to add instructions to deploy SCDF in a production setting with the "released" image as opposed to "latest" tag.

**Acceptance:**
- [Step #7](http://docs.spring.io/spring-cloud-dataflow-server-kubernetes/docs/1.2.1.RELEASE/reference/htmlsingle/#kubernetes-getting-started) could go into the details on how to production deploy SCDF with the right release
- Instructions for minikube setup is available

### LINKED ISSUE #213: Align getting-started chapter to use new config files based on Helm chart templates
We should update the instructions and config files to align with what we use for the Helm chart

### LINKED ISSUE #215: Align server configuration chapter with new config files
The server configuration chapter needs an update and it should be aligned with new config files that match what we use for the Helm chart

### LINKED ISSUE #216: Create new configuration files to align with Helm chart
The server configuration files should be updated to match what we use for the Helm chart

### LINKED ISSUE #217: Deprecate old configuration files
We should keep them since they might be referenced from old documentation. We can remove them with next major release (2.0)

### PR DESCRIPTION
**Pull request by trisberg**
- this is a first step, we'll need to tweak things in a later commit

Resolves #208
Resolves #209
Resolves #213
Resolves #215
Resolves #216
Resolves #217

### CODE REVIEW

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc [hunk 1]
```diff
@@ -170,7 +170,20 @@ You can list all artifacts used by using the following command:
 kubectl get cm,secrets,svc,rc,pod
 ----
 
-=== Server
+== Deployer Properties
+
+The Spring Data Flow Kubernetes Server has several properties you can use to configure the apps that it deploya. You can control the default values to set the `cpu` and `memory` requirements for the pods.  The configuration is controlled by configuration properties under the `spring.cloud.deployer.kubernetes` prefix.  For example you might declare the following section in an `application.properties` file or pass them as command line arguments when starting the Server.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc]: Typo: "deploya"

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc [hunk 2]
```diff
@@ -170,7 +170,20 @@ You can list all artifacts used by using the following command:
...
+
+```
+spring.cloud.deployer.kubernetes.memory=512Mi
+spring.cloud.deployer.kubernetes.cpu=500m
+```
+
+See https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/blob/master/src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployerProperties.java[KubernetesAppDeployerProperties] for more of the supported options.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc]: This link to source 404s.

Comment (sabbyanandan) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc]: Probably you meant to link: https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/blob/master/src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesDeployerProperties.java

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc [hunk 3]
```diff
@@ -170,7 +170,20 @@ You can list all artifacts used by using the following command:
...
+
+Data Flow Server properties that are common across all of the Data Flow Server implementations that concern maven repository settings can also be set in a similar manner.  See the section on Common Data Flow Server Properties for more information.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc]: How about: "Data Flow Server properties that are common across all of the Data Flow Server implementations including the configuration of maven repository settings can be set in a similar manner"

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 4]
```diff
@@ -1,9 +1,45 @@
...
+
+Spring Cloud Data Flow is a toolkit for building data integration and real-time data processing pipelines.
+
+Pipelines consist of Spring Boot apps, built using the Spring Cloud Stream or Spring Cloud Task microservice frameworks. This makes Spring Cloud Data Flow suitable for a range of data processing use cases, from import/export to event streaming and predictive analytics.
+
+This project provides support for using Spring Cloud Data Flow with Kubernetes as the runtime for these pipelines with apps packaged as Docker images.
+
+[[spring-cloud-data-flow-overview]]
+=== Spring Cloud Data Flow
+Spring Cloud Data Flow is a cloud-native orchestration service for composable data microservices on modern runtimes. With Spring Cloud Data Flow, developers can create and orchestrate data pipelines for common use cases such as data ingest, real-time analytics, and data import/export.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: I'd suggest removing the overview sections. We removed it from the core docs, too. Besides, there are 2 different types of core message discussed here. The intro and then this section, that is.

**Review by sabbyanandan — COMMENTED** [review]
@trisberg: Please review the simple change suggestions. Looks great overall!

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc (hunk 1 above)
Comment (trisberg) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc]: fixed

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc (hunk 2 above)
Comment (trisberg) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc]: yep

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc (hunk 3 above)
Comment (trisberg) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc]: sounds good

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc (hunk 4 above)
Comment (trisberg) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc]: Seems to me that not explaining what SCDF is at all before diving in and using it is a bit strange

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc [hunk 5]
```diff
@@ -156,38 +195,108 @@ The security settings in the `scdf-config-kafka.yml` file are:
...
+
+You can get the logs for a specific pod by issuing:
 
 [source,shell]
 ----
-kubectl logs <scdf-pod-name>
+kubectl logs pod <pod-name>
+----
+
+If the pod is continously getting restarted you can add `-p` as an option to see the previous log like:
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc]: "continuously"

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc [hunk 6]
```diff
@@ -156,38 +195,108 @@ The security settings in the `scdf-config-kafka.yml` file are:
...
+[source,shell]
+----
+kubectl get pod -l app=scdf=server
+kubectl logs <scdf-server-pod-name>
 ----
 
 === Streams
 
-The streams apps are deployed with teh stream name followed by the name of the app and for processors and sinks there is also an instance index appended. 
+The streams apps are deployed with the stream name followed by the name of the app and for processors and sinks there is also an instance index appended.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc]: "The stream apps ..."

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc [hunk 7]
```diff
@@ -156,38 +195,108 @@ The security settings in the `scdf-config-kafka.yml` file are:
...
+kubectl get pod -l app=scdf=server
+kubectl logs <scdf-server-pod-name>
 ----
 
 === Streams
 
-The streams apps are deployed with teh stream name followed by the name of the app and for processors and sinks there is also an instance index appended. 
+The streams apps are deployed with the stream name followed by the name of the app and for processors and sinks there is also an instance index appended. 
+
+To see all the pods that are deployed by the Spring Cloud Data Flow server you can specify the label `role=spring=app`:
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc]: `role=spring=app` to `role=spring-app`

**Review by sabbyanandan — APPROVED** [review]
LGTM. Thanks, Thomas!

**Pull request merged**